src/net.cpp

```diff
@@ -2377,6 +2409,15 @@ bool CConnman::Start(CScheduler& scheduler, const Options& connOptions)
         }
     }
 
+    m_anchors = ReadAnchors(GetDataDir() / ANCHORS_DATABASE_FILENAME);
+    if (m_anchors.size() > MAX_BLOCK_RELAY_ONLY_ANCHORS) {
+        m_anchors.resize(MAX_BLOCK_RELAY_ONLY_ANCHORS);
+    }
+
+    if (m_use_addrman_outgoing) {
```

Comment:
I wonder if the call to `ReadAnchors()` should be within this if block. If `ReadAnchors()` is outside the block, then when the node is started with `-noconnect`, not only will the node not connect to the anchors, it'll also throw away the file, so that when it's restarted without `-noconnect`, it won't try to reconnect to the old anchors. Is that what we want?

Reply:
> Is that what we want?

I guess not.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/17428#issuecomment-693556607).

Reply:
hmm, actually this still won't work, since if we start with `-noconnect`, then on shutdown we'll still write a new `anchors.dat` file without any peer addresses.